lib/r10k/module_loader/puppetfile.rb

```diff
@@ -0,0 +1,195 @@
+module R10K
+  module ModuleLoader
+    class Puppetfile
+
+      DEFAULT_MODULEDIR = 'modules'
+      DEFAULT_PUPPETFILE_NAME = 'Puppetfile'
+      DEFAULT_FORGE_API = 'forgeapi.puppetlabs.com'
+
+      attr_accessor :default_branch_override, :environment
+      attr_reader :modules, :moduledir,
+        :managed_directories, :desired_contents, :purge_exclusions
+
+      # @param basedir [String] The path that contains the moduledir &
+      #     Puppetfile by default. May be an environment, project, or
+      #     simple directory.
+      # @param puppetfile [String] The path to the Puppetfile, either an
+      #     absolute full path or a relative path with regards to the basedir.
+      # @param moduledir [String] The path to the moduledir, either an
+      #     absolute full path or a relative path with regards to the basedir.
+      # @param forge [String] The url (without protocol) to the Forge
+      # @param overrides [Hash] Configuration for loaded modules' behavior
+      # @param environment [R10K::Environment] The environment loading may be
+      #     taking place within
```

Comment:
```suggestion
      # @param environment [R10K::Environment] When provided, the environment in which loading
      #     takes place
```
or something like that, to make clear that loading definitely (not may) happens within the environment, when that param is provided.

Reply:
accepted locally